pkg/network/node/networkpolicy.go

```diff
@@ -794,6 +802,35 @@ func (np *networkPolicyPlugin) parsePeerFlows(npns *npNamespace, npp *npPolicy,
 				// checking the source VNID...
 				npp.watchesNamespaces = true
 				peerFlows = append(peerFlows, np.selectNamespaces(peer.NamespaceSelector, dir)...)
+				if np.inMigrationMode {
+					// In SDN live migration mode, we can't use source VNID to
+					// distinguish namespaces for traffic from OVN nodes.
+					// So instead we pretend the rule was a combined
+					// namespaceSelector+podSelector rule with a match-all
+					// podSelector, and generate per-pod-IP match rules.
+					npp.watchesAllPods = true
+					peerFlows = append(peerFlows, np.selectPodsFromNamespaces(peer.NamespaceSelector, &metav1.LabelSelector{}, dir)...)
+
+					// If the host network namespace is selected, Add rules for
+					// the OVN mp0 IP of each node.
+					sel, _ := metav1.LabelSelectorAsSelector(peer.NamespaceSelector)
+					if _, ok := np.nsMatchCache[sel.String()].matches[HostNetworkNamespace]; ok {
+						hsns, err := np.node.osdnInformers.Network().V1().HostSubnets().Lister().List(labels.Everything())
```

Comment:
ok, since this is only for migration mode, and we shouldn't be adding or removing nodes while migrating, you could just precompute the list of mp0-related flows in `Start()` rather than regenerating it every time.

Alternatively... OVS lets you match IPs against _arbitrary_ bitmasks, not just proper CIDR expressions. If `node.networkInfo.ClusterNetworks` is `[{ ClusterCIDR: "10.128.0.0/14", HostSubnetLength: 23 }]`, then the mp0 IPs are exactly the ones that have "`10.128`" in bits 0-14 and "`2`" in bits 23-32 (and anything in bits 15-22). So that's `nw_src=10.128.0.2/0xfffc1f` (where that mask value is `uint32(math.MaxUint32 << (32 - cidrMaskLen)) | (math.MaxUint32 >> hostSubnetLen)`)

Right?